### PR TITLE
Add reusable workflows for go, build go action

### PR DESCRIPTION
**Pull request by viccuad**
Tested with psp-hostpaths-policy:
https://github.com/viccuad/psp-hostpaths-policy/actions/runs/1746731647
https://github.com/viccuad/psp-hostpaths-policy/actions/runs/1746731639
See https://github.com/kubewarden/hostpaths-psp-policy/pull/3.

Fixes: https://github.com/kubewarden/github-actions/issues/2.

### CODE REVIEW

**Comment by viccuad** [comment]
Done in https://github.com/kubewarden/github-actions/pull/7/commits/f7b61d678ba9500f1f2bc53c4ddb08e40880718f.

File: .github/workflows/reusable-release-policy-go.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+on:
+  workflow_call:
+    inputs:
+      oci-target:
+        type: string
+        required: true
+    secrets:
+      workflow-pat:
+        description: "Github Personal Access Token (PAT) that can trigger workflow-dispatch"
+        required: true
```
Comment (jvanz) [inline .github/workflows/reusable-release-policy-go.yml:12]: I think you do not need this secret. AFAICS, this token is used to notify the policy hub. However, I believe you can use the $secrets.GITHUB_TOKEN$ for that as well. 
https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow

**Review by flavio — APPROVED** [review]
LGTM

File: .github/workflows/reusable-release-policy-go.yml (hunk 1 above)
Comment (flavio) [reply .github/workflows/reusable-release-policy-go.yml:12]: this is not possible in this case, we have to trigger an action on another repository. the github token has privileges only inside of the repo that owns the action.

In this case a policy needs to trigger an action inside of the policy-hub repository.

Comment (viccuad) [reply .github/workflows/reusable-release-policy-go.yml:12]: I thought the same, but I think it cannot be :/. From https://github.com/marketplace/actions/workflow-dispatch#token, "NOTE. The automatically provided token e.g. ${{ secrets.GITHUB_TOKEN }} can not be used, GitHub prevents this token from being able to fire the workflow_dispatch".

But now I'm wondering if we are really doing a workflow_dispatch...

Comment (flavio) [reply .github/workflows/reusable-release-policy-go.yml:12]: we are, by triggering the dispatch via github's API

Comment (viccuad) [reply .github/workflows/reusable-release-policy-go.yml:12]: Meerging then.

Comment (jvanz) [reply .github/workflows/reusable-release-policy-go.yml:12]: :+1:

**Pull request merged**